### ISSUE TITLE
Add statistical annotations (CIs, Cohen's d) on charts

### ISSUE THREAD

**Issue by xcoder-es** [Nomos-N4s/nomos#77]
## Task
Annotate dashboard charts with statistical information: confidence intervals, effect sizes, and significance indicators.

## Implementation
1. **Benchmark tab**: Add Cohen's d value and significance stars to each comparison row in the tables
2. **RL Training tab**: Add 95% CI bands to reward curves (use bootstrap or std error)
3. **All tabs**: Add a 'Show Statistics' toggle that reveals annotations
4. Add a summary statistics section at the top of each tab

## Altair Implementation
- Use alt.Layer with area marks for CI bands
- Use alt.condition for significance coloring
- Add tooltips showing statistical values

## Acceptance Criteria
- CI bands visible on reward curves when toggled on
- Effect sizes shown on benchmark comparison tables
- Significance indicators (p < 0.05, p < 0.01, p < 0.001)

**Comment by felista19** [comment]
Hi! I'd like to work on this issue.

**Comment by xcoder-es** [comment]
> Hi! I'd like to work on this issue.

Sure! Go for it. 

Please just remember to request a PR review by manually adding me as one of the PR reviewers.

Also, update the ticket status accordingly.

Thanks!

**Comment by felista19** [comment]
I've started implementing the dashboard statistics enhancements.

Current progress:
Added a global "Show Statistics" toggle to the dashboard.
Updated the dashboard tabs to accept the new statistics toggle.
While working on the chart annotations, I noticed the repository doesn't include sample benchmark/RL result files (benchmark_results.json and results/rl/*) needed to verify confidence intervals and statistical overlays. Are there sample result files available, or is there an expected schema I should target?

**Comment by xcoder-es** [comment]
Good question — the result files are gitignored (`results/*` is in `.gitignore`), so they won't appear when you clone. You'll need to generate them locally:

```
python -m src.governance.runner all --baselines --steps 100 --seeds 5
```

That produces `results/benchmark_results.json` and CSVs. For reference, here's the JSON schema so you can work without the files:

**aggregates** — one entry per strategy/scenario pair:
`{strategy, scenario, num_seeds, mean_reward, std_reward, mean_deadlocks, mean_violations, ci_lower, ci_upper}`

**effect_sizes** — governance vs each baseline per scenario:
`{scenario, governance_vs, cohens_d, cohens_d_ci, cohens_d_se, mannwhitney_u, p_value_raw, p_value_corrected, p_value_holm, significant, significant_holm, interpretation}`

**reward_hacking_episodes** — step-level detections:
`{step, reward_spike, violation_count, strategy, scenario, seed}`

The existing `benchmarks_tab.py` loads this JSON via `json.load()` and renders the aggregates table — you can follow that pattern. For RL reward curves with CI bands, you can reuse the `_bootstrap_ci()` function from `src/governance/benchmarks/analysis.py`.

**Comment by felista19** [comment]
@xcoder-es 
 Hi! Quick update on Issue #77
 1 Implemented the benchmark dashboard enhancements.
2 Added a consolidated 95% CI column to the per-scenario results table.
3 Added a Statistical Analysis section with Cohen's d, interpretation, significance markers, and significance status.

Currently working on the RL Training confidence interval visualization.
 Quick question: _bootstrap_ci() computes confidence intervals from a list of values, but the dashboard currently loads governed_metrics.csv and ungoverned_metrics.csv, which appear to contain one reward value per training step. For the 95% CI bands should I compute them from multiple per-seed training CSVs, or is there another intended data source?

**Comment by xcoder-es** [comment]
@felista19  Great progress! 

For the RL Training CI visualization: `_bootstrap_ci()` expects a list of values to compute confidence bounds from. Since a single CSV (one reward per step) gives you only one observation per step, there's no variance to bootstrap. You need multiple seeds.

The RL training script saves per-seed results in separate files (check `results/rl/` after running `python -m src.governance.runner adversary` with multiple seeds). The pattern is:

1. Group all seed CSVs for the same condition (governed vs ungoverned)
2. At each training step, collect the reward values across seeds
3. Call _bootstrap_ci(rewards_at_step) to get the CI band at that step
4. Plot the mean as the main trace and the CI band as a shaded region

If only one seed is available (single CSV), skip the CI band and just plot the raw trace. You can detect this by checking `len(seed_files) < 2`.

For the per-seed file loading, the `dashboard/rl_tab.py` already has a helper that discovers CSVs, look for how it loads `governed_metrics.csv` and extend it to glob for seed-patterned files.

**Comment by xcoder-es** [comment]
@felista19 Quick heads-up — RISO525 has picked up #78 (side-by-side comparison mode) which touches the same dashboard tabs (`rl_tab.py` and `benchmarks_tab.py`). Your statistical annotations (#77) and their comparison mode are complementary, but coordinating on data-loading helpers and chart utility functions would save rebase pain later.

I've started a Discussion thread where you two can sync up on the shared bits: https://github.com/xcoder-es/governance-layer/discussions/1

Your toggle pattern ("Show Statistics") and their comparison toggle should compose cleanly since they gate on different conditions.
